src/main/java/org/elasticsearch/index/mapper/internal/IdFieldMapper.java

```diff
@@ -144,6 +146,8 @@ protected IdFieldMapper(String name, String indexName, float boost, FieldType fi
         super(new Names(name, indexName, indexName, name), boost, fieldType, docValues, Lucene.KEYWORD_ANALYZER,
                 Lucene.KEYWORD_ANALYZER, postingsProvider, docValuesProvider, null, null, fieldDataSettings, indexSettings);
         this.path = path;
+        this.writePre20Settings = indexSettings.equals(ImmutableSettings.EMPTY) == false &&
```

Comment:
It is, because `Version.indexCreated` throws an exception if it cannot find the version setting (and the default constructor above uses empty settings).